app/services/coupons/update_service.rb

```diff
@@ -18,12 +18,23 @@ def call
       coupon.expiration_at = params[:expiration_at] if params.key?(:expiration_at)
 
       @limitations = params[:applies_to]&.to_h&.deep_symbolize_keys || {}
+      coupon_already_applied = coupon.applied_coupons.exists?
 
-      unless coupon.applied_coupons.exists?
+      unless coupon_already_applied
         if !plan_identifiers.nil? && plans.count != plan_identifiers.count
           return result.not_found_failure!(resource: 'plans')
         end
 
+        if !billable_metric_identifiers.nil? && billable_metrics.count != billable_metric_identifiers.count
+          return result.not_found_failure!(resource: 'billable_metrics')
+        end
+
+        if (coupon.billable_metrics.exists? && plans.present?) ||
+         (coupon.plans.exists? && billable_metrics.present?) ||
+         (billable_metrics.present? && plans.present?)
```

Comment:
Layout/MultilineOperationIndentation: Align the operands of a condition in an `if` statement spanning multiple lines.